venue/tasks.py

```diff
@@ -54,110 +70,128 @@ def get_expected_links(code):
+            for post in posts:
+                message_ids.append(post['message_id'])
+                post_check = ForumPost.objects.filter(
                     forum_profile=forum_profile,
-                    forum_rank=forum_profile.forum_rank,
                     topic_id=post['topic_id'],
                     message_id=post['message_id'],
-                    unique_content_length=post['content_length'],
-                    timestamp=post['timestamp']
                 )
-                forum_post.save()
-        # Check for post deletion
-        deleted_posts = ForumPost.objects.filter(
-            forum_profile=forum_profile,
-            matured=False,
-            timestamp__gte=posts_scrape_start
-        ).exclude(
-            message_id__in=message_ids
-        )
-        for post in deleted_posts:
-            post.credited = False
-            post.monitoring = False
-            post.save()
-        # Update tracked posts
-        for post in tracked_posts:
-            forum_post = ForumPost.objects.get(
-                message_id=post,
-                forum_profile=forum_profile
+                if not post_check.exists():
+                    forum_post = ForumPost(
+                        user_profile=forum_profile.user_profile,
+                        forum_profile=forum_profile,
+                        forum_rank=forum_profile.forum_rank,
+                        topic_id=post['topic_id'],
+                        message_id=post['message_id'],
+                        unique_content_length=post['content_length'],
+                        timestamp=post['timestamp']
+                    )
+                    forum_post.save()
+            # Check for post deletion
+            deleted_posts = ForumPost.objects.filter(
+                forum_profile=forum_profile,
+                matured=False,
+                timestamp__gte=posts_scrape_start
+            ).exclude(
+                message_id__in=message_ids
             )
-            forum_post.save()
-        # Update the forum_profile's last scrape timestamp
-        forum_profile.last_scrape = timezone.now()
-        forum_profile.save()
+            for post in deleted_posts:
+                post.credited = False
+                post.monitoring = False
+                post.save()
+            # Update tracked posts
+            for post in tracked_posts:
+                forum_post = ForumPost.objects.get(
+                    message_id=post,
+                    forum_profile=forum_profile
+                )
+                forum_post.save()
+            # Update the forum_profile's last scrape timestamp
+            forum_profile.last_scrape = timezone.now()
+            forum_profile.save()
+    except MaxRetriesExceededError:
+        message = 'Consistent forum profile scraping error: %s' % forum_profile_id
+        log.error(message)
+        # execute a task that will trigger a rollbar error
+        report_error.delay(message)
```

Comment:
Can you add more context here? Only forum_profile_id it's not enough, I'd like to have a context of the error. 

And I don't see a reason to send exceptions to the `dnalogs` if we already have them in the `rollbar`. As for me `dnalogs` more for the info and to trace requests.

Reply:
@alexdashkov  I'll be adding more log messages and add more context/details to the logs.

The rationale for sending error logs to LogDNA is for completeness sake and for easier post-mortem analysis. We can get more details about the error in Rollbar (e.g. complete error traceback), but we should at least have a log message in LogDNA that the error occurred, so we don't have to keep switching between the two to determine the timing of the errors occur.